tough/src/lib.rs

```diff
@@ -874,10 +887,105 @@ fn load_targets<T: Transport>(
 
     // Now that everything seems okay, write the targets file to the datastore.
     datastore.create("targets.json", &targets)?;
+    if let Some(delegations) = &mut targets.signed.delegations {
+        load_delegations(
+            transport,
+            snapshot,
+            metadata_base_url,
+            max_targets_size,
+            delegations,
+            &datastore,
+        )?;
+    }
 
     Ok(targets)
 }
 
+//Follow the paths of delegations starting with the top level targets.json delegation
+fn load_delegations<T: Transport>(
+    transport: &T,
+    snapshot: &Signed<Snapshot>,
+    metadata_base_url: &Url,
+    max_targets_size: u64,
+    delegation: &mut Delegations,
+    datastore: &Datastore<'_>,
+) -> Result<()> {
+    let mut delegated_roles: HashMap<String, Option<Signed<crate::schema::Targets>>> =
+        HashMap::new();
+    for del_role in &delegation.roles {
+        //find the role file metadata
+        let role_meta = snapshot
+            .signed
+            .meta
+            .get(&format!("{}.json", &del_role.name))
+            .unwrap();
+
+        let path = format!("{}.json", &del_role.name);
```

Comment:
Same question as earlier: does consistent snapshots affect the filename by prepending a version for these?

```rust
        let path = format!("{}.json", &del_role.name);
```

Reply:
That is true, and will be updated in incoming PR along with write and sign capabilities.